crates/bevy_sprite/src/mesh2d/mesh.rs

```diff
@@ -178,29 +184,57 @@ bitflags::bitflags! {
     }
 }
 
+pub struct RenderMesh2dInstance {
+    pub transforms: Mesh2dTransforms,
+    pub mesh_asset_id: AssetId<Mesh>,
+    pub material_bind_group_id: Material2dBindGroupId,
+    pub automatic_batching: bool,
+}
+
+#[derive(Default, Resource, Deref, DerefMut)]
+pub struct RenderMesh2dInstances(PassHashMap<Entity, RenderMesh2dInstance>);
+
+#[derive(Component)]
+pub struct Mesh2d;
+
 pub fn extract_mesh2d(
     mut commands: Commands,
-    mut previous_len: Local<usize>,
-    query: Extract<Query<(Entity, &ViewVisibility, &GlobalTransform, &Mesh2dHandle)>>,
+    mut render_mesh_instances: ResMut<RenderMesh2dInstances>,
+    query: Extract<
+        Query<(
+            Entity,
+            &ViewVisibility,
+            &GlobalTransform,
+            &Mesh2dHandle,
+            Has<NoAutomaticBatching>,
```

Comment:
When is `NoAutomaticBatching` added to a main world entity?

Reply:
Is this rhetorical and suggesting I should document it?

Main world entities that have morphs or skins will have it added. Otherwise users can choose to add it if they want to opt out of batching.

Reply:
For `mesh2d/mesh.rs` here I don't think morphs/skins are relevant.

But what I was asking is: the idea is to make automatic batching user-controllable on an entity basis. But why? Isn't it a property of the specific material? If so, is a separate component justified? I mean, non-blocking, but worth thinking about.

Reply:
The component was introduced in the batching PR. The point was to allow people to opt-out of automatic batching and instancing if for some reason it was causing performance issues, or if they could do their own batching/instancing in some way that they want to. It's an escape hatch for user control.